### PR TITLE
Implement Location Servlet

### PR DESCRIPTION
**Pull request by nssafir**
Implement a location servlet that uses Google's Geocoder API to get important features (ex. street name, city, ZIP code) from a user's location coordinates. This will later fed into the Google Civic API to get users in touch with their local politicians. Later, I will need to implement a doPost() method with Google's geolocation API to get the user's coordinates if they consent.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/LocationServlet.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    * Converts json response from Geolocation API to HashMap with important geographic characteristics.
+    */
+    private void jsonToMap(String jsonString) {
+      JsonObject responseObject = new JsonParser().parse(jsonString).getAsJsonObject();
+      try {
+        JsonArray resultsArray = responseObject.getAsJsonArray("results");
+        JsonArray addressArray = resultsArray.get(0).getAsJsonObject().getAsJsonArray("address_components");
+        JsonObject entry;
+        String name, geoType;
+        for (int i = 0; i < addressArray.size(); i++){
```
Comment (juanuribe28) [inline src/main/java/com/google/sps/servlets/LocationServlet.java:77]: [Optional] A comment here to explain the logic would be useful.

File: src/main/java/com/google/sps/servlets/LocationServlet.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          entry = addressArray.get(i).getAsJsonObject();
+          name = entry.getAsJsonPrimitive("long_name").getAsString();
+          geoType = entry.getAsJsonArray("types").get(0).getAsString();
+          if (geoType.equals("locality")) {
```
Comment (juanuribe28) [inline src/main/java/com/google/sps/servlets/LocationServlet.java]: I know switch statements are not ideal, but this seems like a a good place to use one.

File: src/test/java/com/google/sps/LocationTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.google.sps;
```
Comment (juanuribe28) [inline src/test/java/com/google/sps/LocationTest.java]: I would leave this file out of this PR, as it doesn't contain any tests.

File: src/main/java/com/google/sps/servlets/LocationServlet.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        JsonArray resultsArray = responseObject.getAsJsonArray("results");
+        JsonArray addressArray = resultsArray.get(0).getAsJsonObject().getAsJsonArray("address_components");
+        JsonObject entry;
+        String name, geoType;
+        // Extract useful location informatin from JSON array.
+        for (int i = 0; i < addressArray.size(); i++){
+          entry = addressArray.get(i).getAsJsonObject();
+          name = entry.getAsJsonPrimitive("long_name").getAsString();
+          geoType = entry.getAsJsonArray("types").get(0).getAsString();
+          switch (geoType) {
```
Comment (juanuribe28) [inline src/main/java/com/google/sps/servlets/LocationServlet.java:81]: [Nit] It requires a default label, even if it is empty.

**Review by juanuribe28 — CHANGES_REQUESTED** [review]

**Review by juanuribe28 — APPROVED** [review]

**Review by seangilpin — APPROVED** [review]

**Pull request merged**